[Inductor] We re-enable the batch_fusion and group_fusion flags in order not to disturb the current production model implementation

mengluy0125 opened the pull request:
Summary:
We did two things:
1. We add back the batch_fusion and group_fusion flags to keep the current production model implementation

2. We tell batch and group fusion in the post grad since group need fbgemm.

Test Plan:
```
buck2 test 'fbcode//mode/dev-nosan' fbcode//caffe2/test/inductor:group_batch_fusion
```
Buck UI: https://www.internalfb.com/buck2/13d152d2-5d4d-4c7a-ab88-51f8e8218942
Test UI: https://www.internalfb.com/intern/testinfra/testrun/1125900253044737
Network: Up: 376KiB  Down: 44KiB  (reSessionID-c508aedc-8cc2-434a-8c17-bbe075a05562)
Jobs completed: 17. Time elapsed: 1:23.1s.
Cache hits: 0%. Commands: 1 (cached: 0, remote: 0, local: 1)
Tests finished: Pass 6. Fail 0. Fatal 0. Skip 0. Build failure 0

Differential Revision: D51695982




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

jackiexu1992 reviewed (approved):
Accept to unblock. Need to figure out a long term plan for flag+configs.

mengluy committed:
[Inductor] We re-enable the batch_fusion and group_fusion flags in order not to disturb the current production model implementation (#114841)

Summary:

We did two things:
1. We add back the batch_fusion and group_fusion flags to keep the current production model implementation

2. We tell batch and group fusion in the post grad since group need fbgemm.

Test Plan:
```
buck2 test 'fbcode//mode/dev-nosan' fbcode//caffe2/test/inductor:group_batch_fusion
```
Buck UI: https://www.internalfb.com/buck2/e2b851cd-1797-44f1-9a1e-a9f3875232c5
Test UI: https://www.internalfb.com/intern/testinfra/testrun/15481123731700522
Network: Up: 8.8MiB  Down: 61MiB  (reSessionID-a6e9167f-884d-4bcf-8088-f079d01e45c8)
Jobs completed: 189398. Time elapsed: 3:21.7s.
Cache hits: 99%. Commands: 76982 (cached: 76977, remote: 4, local: 1)
Tests finished: Pass 7. Fail 0. Fatal 0. Skip 0. Build failure 0

Reviewed By: yanboliang, jackiexu1992

Differential Revision: D51695982
